postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -180,6 +180,10 @@ def __init__(self, check, config, shutdown_callback):
         )
         if collection_interval <= 0:
             collection_interval = DEFAULT_COLLECTION_INTERVAL
+        
+        # if regular samples is disabled, only need to collect as often as activity is sampled
+        if not is_affirmative(config.statement_samples_config.get('enabled', True)):
```

Comment:
is there some extra logic we can add [in either of these tests](https://github.com/DataDog/integrations-core/blob/master/postgres/tests/test_statements.py#L1377-L1419) to exercise this logic?